### PR TITLE
CI/TYP: Window typing followup

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

I believe https://github.com/pandas-dev/pandas/pull/40293 may have caused the current typing failures in the CI

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Thanks for the follow-up!

File: pandas/core/window/expanding.py [hunk 1]
```diff
@@ -589,7 +586,7 @@ def quantile(
     )
     def cov(
         self,
-        other: Optional[Union[np.ndarray, FrameOrSeriesUnion]] = None,
+        other: Optional[FrameOrSeriesUnion] = None,
```
Comment (jorisvandenbossche) [inline pandas/core/window/expanding.py:589]: How is this change correct? Or is the docstring not correct? (which says that other can also be an ndarray)

**Review by jorisvandenbossche — COMMENTED** [review]
Merge this to get master green again, but still a question about the actual change

File: pandas/core/window/expanding.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/expanding.py:589]: At least on master, passing a numpy array raises

```
In [1]: pd.Series(range(10)).expanding().cov(np.arange(10))
Exception: Input arrays must be of the same type!
```
as well there's a test, `test_different_input_array_raise_exception` that checks that an `Exception` is raised with a numpy array.

So I can follow up fixing the documentation

**Pull request merged**